### ISSUE TITLE
body.outputStream.on is not a function

### ISSUE THREAD

**Issue by waldiTM** [Azure/azure-xplat-cli#2230]
azure vm image create using node 4.1.1 and azure-cli 0.9.10 fails:
```
verbose: stack TypeError: body.outputStream.on is not a function
    at /usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:431:35
    at /usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:557:5
    at SharedKey.signRequest (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/signing/sharedkey.js:84:3)
    at BlobService.StorageServiceClient._buildRequestOptions (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:539:27)
    at operation (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:262:10)
    at /usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:470:9
    at BlobService.filter (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:826:7)
    at BlobService.StorageServiceClient._performRequest (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:453:8)
    at BlobService.StorageServiceClient.performRequestOutputStream (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:193:8)
    at BlobService._createPages (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/services/blob/blobservice.js:4652:10)
```

**Comment by yugangw-msft** [comment]
@waldiTM,  if you remove the /usr/local/.streamline, does that help?

**Comment by yugangw-msft** [comment]
closing, please feel free reopen it with more information.

**Comment by savs** [comment]
I'm seeing the same error.

azure vm image create -vv -u https://mything.blob.core.windows.net/vm-images/thing.vhd --location "West Europe" --os linux -p 32 -f thing mydisk.vhd

silly:   Starting to calculate MD5 hash for input file (use -m to skip it)
error:   body.outputStream.on is not a function
verbose: stack TypeError: body.outputStream.on is not a function
    at /usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:431:35
    at /usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:557:5
    at SharedKey.signRequest (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/signing/sharedkey.js:84:3)
    at BlobService.StorageServiceClient._buildRequestOptions (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:539:27)
    at operation (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:262:10)
    at /usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:470:9
    at BlobService.filter (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:826:7)
    at BlobService.StorageServiceClient._performRequest (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:453:8)
    at BlobService.StorageServiceClient.performRequestOutputStream (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:193:8)
    at BlobService._createPages (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/services/blob/blobservice.js:4652:10)
silly:   { stack: 'TypeError: body.outputStream.on is not a function\n    at /usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:431:35\n    at /usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:557:5\n    at SharedKey.signRequest (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/signing/sharedkey.js:84:3)\n    at BlobService.StorageServiceClient._buildRequestOptions (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:539:27)\n    at operation (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:262:10)\n    at /usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:470:9\n    at BlobService.filter (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:826:7)\n    at BlobService.StorageServiceClient._performRequest (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:453:8)\n    at BlobService.StorageServiceClient.performRequestOutputStream (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/common/services/storageserviceclient.js:193:8)\n    at BlobService._createPages (/usr/local/lib/node_modules/azure-cli/node_modules/azure-storage/lib/services/blob/blobservice.js:4652:10)', message: 'body.outputStream.on is not a function' }

**Comment by EgoAleSum** [comment]
I'm experiencing the same error with Node 5.2.0 on Windows 10 x64 and azure-cli 0.9.13. A customer of mine is reporting the same issue from a Linux client.

**Comment by yugangw-msft** [comment]
add @sauryadas @huangpf

**Comment by yugangw-msft** [comment]
@yaxia , is this a know node 4+ issues from azure-storage?

**Comment by huangpf** [comment]
I also think this is a storage client issue, as shown from the logs.

@yugangw-msft Is ```5.2.0``` the official version that works with Azure CLI?
@EgoAleSum Is it feasible for you to try it with a lower node version, like: ```v0.12.7```?

**Comment by yaxia** [comment]
It is the same issue as https://github.com/Azure/azure-xplat-cli/issues/2168. The root cause is [here](https://github.com/wastoresh/azure-xplat-cli/blob/dev/lib/commands/asm/iaas/upload/pageBlob.js#L732). When it tries to upload a dynamic disk, the thrid parameter of ```blobService.createPagesFromStream``` is supposed to be a stream, but it passes an buffer instead, which doesn't has an ```on``` listener registered.

**Comment by EgoAleSum** [comment]
Thanks all for responding quickly.

If it's a problem with dynamically allocated VHDs, at least I can give a quick workaround (fixed-size VHDs) to my customer (I'm an Azure TSP for MS Canada btw). However, would be nice to have this bug fixed...

@huangpf I can try with 0.12 tomorrow (from the office, where the connection to Azure is faster :) ) if necessary... However #2168 seems to say that's also on 0.12

**Comment by huangpf** [comment]
Thanks @yaxia 

@EgoAleSum - Yes, I can reproduce the issue by uploading dynamic VHD, so it's not about the aforementioned version issue. Working on the fix now.

**Comment by alex2006hw** [comment]
Is the fix in place?  I am waiting on a solution.  I have been using statically allocated VHDs as work-around but, I have a customer who need dynamic VHDs.

**Comment by anuchandy** [comment]
@EgoAleSum @alex2006hw @waldiTM  you could use the utility below to upload VHDs (dynamic, fixed) while our storage team fixes this issue
https://github.com/Microsoft/azure-vhd-utils-for-go

**Comment by z3ntu** [comment]
I am experiencing the same issue with a dynamic VHD.

**Comment by yaxia** [comment]
Let me take the code of 0.9.19 as a reference. Please look into the code here: https://github.com/Azure/azure-xplat-cli/blob/v0.9.19-March2016/lib/commands/asm/iaas/upload/pageBlob.js#L732 It should pass a stream instead of a buffer when you call `blobService.createPagesFromStream`. @huangpf

**Comment by mcronce** [comment]
Is there any news on this guy?  I just ran into the same issue and unfortunately with the size of disk I'm working with, a fixed-size VHD is not an option :(

**Comment by anuchandy** [comment]
@mcronce CLI does not support dynamic VHD, please see https://azure.microsoft.com/en-us/documentation/articles/virtual-machines-linux-classic-create-upload-vhd/, as mentioned in this doc you have to use go tool for the same.

**Comment by CalvinHartwell** [comment]
+1 facing the same issue, thanks for the useful replies; will share this with my customers.

**Comment by stefannilsson** [comment]
+1
